From ce11f46b015e494f423fee01e8e506bb4671451c Mon Sep 17 00:00:00 2001
From: Jonathan M Hsieh <jmhsieh@apache.org>
Date: Wed, 24 Sep 2014 06:40:04 -0700
Subject: [PATCH 40/63] HBASE-12080 Shorten the run time of integration test by default when using mvn failsafe:integration-test (Jingcheng Du)

Conflicts:

	hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestIngestWithMOB.java

Reason: MOB feature
Author: Jingcheng Du
Ref: CDH-19420
---
 .../hadoop/hbase/IntegrationTestIngestWithMOB.java |   13 ++++++++++---
 1 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestIngestWithMOB.java b/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestIngestWithMOB.java
index e7caade..c48e509 100644
--- a/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestIngestWithMOB.java
+++ b/hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestIngestWithMOB.java
@@ -31,6 +31,7 @@ import org.apache.hadoop.hbase.util.LoadTestDataGeneratorWithMOB;
 import org.apache.hadoop.hbase.util.LoadTestTool;
 import org.apache.hadoop.util.ToolRunner;
 import org.junit.Assert;
+import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
 /**
@@ -44,9 +45,10 @@ public class IntegrationTestIngestWithMOB extends IntegrationTestIngest {
   public static final String THRESHOLD = "threshold";
   public static final String MIN_MOB_DATA_SIZE = "minMobDataSize";
   public static final String MAX_MOB_DATA_SIZE = "maxMobDataSize";
-  private int threshold = 100 * 1024; //100KB
-  private int minMobDataSize = threshold * 4 / 5; //80KB
-  private int maxMobDataSize = threshold * 50; // 5MB
+  private int threshold = 1024; // 1KB
+  private int minMobDataSize = 512; // 512B
+  private int maxMobDataSize = threshold * 5; // 5KB
+  private static final long JUNIT_RUN_TIME = 2 * 60 * 1000; // 2 minutes
 
   //similar to LOAD_TEST_TOOL_INIT_ARGS except OPT_IN_MEMORY is removed
   protected String[] LOAD_TEST_TOOL_MOB_INIT_ARGS = {
@@ -98,6 +100,11 @@ public class IntegrationTestIngestWithMOB extends IntegrationTestIngest {
     }
   }
 
+  @Test
+  public void testIngest() throws Exception {
+    runIngestTest(JUNIT_RUN_TIME, 100, 10, 1024, 10);
+  };
+
   @Override
   protected void initTable() throws IOException {
     int ret = loadTool.run(getArgsForLoadTestToolInitTable());
-- 
1.7.0.4

